camel/toolkits/video_toolkit.py

```diff
@@ -0,0 +1,583 @@
+                        break
+
+                    chunk_duration = self._get_video_length_from_file(
+                        chunk_filename
+                    )
+                    video_length += chunk_duration
+                    self._chunk_durations.append(chunk_duration)
+                    chunk_index += 1
+            else:
+                video_path = os.path.join(
+                    self.video_download_path,
+                    f'full_video{self.video_extension}',
+                )
+                if os.path.exists(video_path):
+                    video_length = self._get_video_length_from_file(video_path)
+
+        if video_length == 0:
+            ydl_opts = {
+                'quiet': True,
+                'no_warnings': True,
+                'cookiefile': self.cookies_path,
+            }
+
+            try:
+                with self.yt_dlp.YoutubeDL(ydl_opts) as ydl:
+                    info_dict = ydl.extract_info(
+                        self.video_url, download=False
+                    )
+                    video_length = info_dict.get('duration', 0)
+            except Exception as e:
+                print(f"Error retrieving video length from network: {e}")
+                video_length = 0
+
+        return video_length
+
+    def _get_video_length_from_file(self, file_path: str) -> int:
+        try:
+            result = subprocess.run(
+                [
+                    "ffprobe",
+                    "-v",
+                    "error",
+                    "-show_entries",
+                    "format=duration",
+                    "-of",
+                    "default=noprint_wrappers=1:nokey=1",
+                    file_path,
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+            duration = float(result.stdout)
+            return int(duration)
+        except Exception as e:
+            print(f"Error calculating video length from file: {e}")
+            return 0
+
+    def get_video_bytes(self) -> bytes:
+        r"""Returns the video bytes for the downloaded video. If the video was
+        downloaded in chunks, the chunks will be read and concatenated.
+
+        Returns:
+            bytes: The video file content in bytes.
+        """
+        if self.chunk_duration > 0:
+            video_bytes = b""
+            chunk_index = 0
+
+            while True:
+                chunk_filename = os.path.join(
+                    self.video_download_path,
+                    f'video_chunk_{chunk_index}{self.video_extension}',
+                )
+                if not os.path.exists(chunk_filename):
+                    break
+
+                with open(chunk_filename, "rb") as chunk_file:
+                    video_bytes += chunk_file.read()
+
+                chunk_index += 1
+
+            if chunk_index == 0:
+                self.download_video()
+                return self.get_video_bytes()
```

Comment:
personally I don't suggest the recursive call here, have 2 issues:
1.  If `self.download_video()` doesn't properly download the video or fails for any reason, the method could end up calling itself recursively indefinitely
2. recursive calls within the same method makes the code harder to understand and maintain. It may not be immediately clear why the method calls itself